### PR TITLE
Hide irrelevant test jobs from deployment API payload

### PR DESCRIPTION
**Pull request by jonmv**
@smorgrav please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java [hunk 1]
```diff
@@ -194,8 +194,14 @@ public boolean hasFailures() {
         return instances.build();
     }
 
-    /** The step status for all steps in the deployment spec of this, in the same order as in the deployment spec. */
-    public List<StepStatus> allSteps() { return allSteps; }
+    /** The step status for all relevant steps in the deployment spec of this, in the same order as in the deployment spec. */
+    public List<StepStatus> allSteps() {
+        List<JobId> firstTestJobs = List.of(firstDeclaredOrElseImplicitTest(systemTest),
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:199]: Only these test jobs will be triggered automatically. Undeclared jobs may still be triggered by users, though, but that likely won't happen if they aren't shown here :)

Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:199]: When will an implicit job be present here?

**Review by smorgrav — APPROVED** [review]
I trust you on this, albeit I'm not sure I understand the implications on when the implicit jobs are included or not

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:199]: When there aren't any explicit ones.

**Comment by jonmv** [comment]
If there is a declared job of a specific type, the first of these will be chosen as the test job for instances without a declared job, which need that test run; in this case, no implicit job will run, i.e., not be shown here. 
If there are no declared jobs of a type, the same (first) instance will be used to run that test job whenever needed; in this case, that job will be among these `firstTestJobs`, and the step with that implicit job will not be filtered out.

**Comment by jonmv** [comment]
This is way too complicated.

**Pull request merged**